src/state/sagas/datasets/DeleteDataset/DeleteDataset.js

```diff
@@ -23,13 +22,16 @@ export function* deleteDataset(action) {
     const filteredDatasets = datasets.filter((dataset) => dataset.id !== datasetId);
     const newSelectedDatasetIndex = filteredDatasets.findIndex((dataset) => dataset.id === selectedDatasetId);
```

Comment:
When we delete a currently selected dataset, the first in the list is automatically selected, it's assured by `useCurrentDataset` hook that returns the first dataset if the index is wrong. In this saga, we could add a condition to not use `SET_CURRENT_DATASET_INDEX` if the index is -1, what do you think ? 

Perhaps, I see a more tricky problem : when `useCurrentDataset` selects the first dataset, it doesn't reset the index, so we have an erroneous one in Redux. It could be a source of problems somehow or not ? :thinking:

Reply:
Regarding the first point, maybe the best solution would be to add protection in the reducer, in the `SET_CURRENT_DATASET_INDEX` case, to prevent setting a negative index or an index that is greater than or equal to the number of datasets. This way, we don't have to add specific checks (index != -1) in all sagas and dispatchers.

For the second issue, we should probably set the index too in `useCurrentDataset`, yes :+1: